tools/db_crashtest.py

```diff
@@ -121,14 +121,32 @@
 }
 
 _TEST_DIR_ENV_VAR = 'TEST_TMPDIR'
+_DISK_TEMP_DIR_ENV_VAR = "DISK_TEMP_DIR"
+_HOME_ENV_VAR = "HOME"
+
+
+def need_direct_io_support(params):
+    return (params["use_direct_reads"] == 1 or \
+            params["use_direct_io_for_flush_and_compaction"] == 1) and \
+            params["mmap_read"] == 0
 
 
 def get_dbname(test_name):
+    test_dir_name = "rocksdb_crashtest_" + test_name
     test_tmpdir = os.environ.get(_TEST_DIR_ENV_VAR)
     if test_tmpdir is None or test_tmpdir == "":
-        dbname = tempfile.mkdtemp(prefix='rocksdb_crashtest_' + test_name)
+        dbname = tempfile.mkdtemp(prefix=test_dir_name)
```

Comment:
This will leave the tmp directory around if dbname is overriden later with another directory for direct IO.

Reply:
if it's overriden, `shutil.rmtree` will delete the dbname

Reply:
This comment is not relevant anymore.